subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6711,6 +6719,10 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 	if ((le_event_mask & BT_EVT_MASK_LE_PER_ADVERTISING_REPORT) && accept &&
 	    ((data_len_total - data_len) < CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX)) {
 
+		if (data_len_total > data_len) {
+			tx_pwr = BT_HCI_LE_ADV_TX_POWER_NO_PREF;
+		}
```

Comment:
The data length relevance towards the tx_pwr is not 100% clear here. I assume that this is implicitly checking if the tx power is in the advertising report, right? 

Can you please add a comment explaining this part?